services/src/main/java/org/fao/geonet/api/records/editing/EditUtils.java

```diff
@@ -91,7 +88,7 @@ protected static void addMissingGeoNetRef(Element element) {
         for (Object descendant : list) {
             if (descendant instanceof Element) {
                 Element e = (Element) descendant;
-                if (e.getName() != Edit.RootChild.ELEMENT
+                if (!e.getName().equals(Edit.RootChild.ELEMENT)
```

Comment:
So this was indeed wrong since https://github.com/geonetwork/core-geonetwork/commit/1a4c8d1e61d57d431c3a457d0191a6f8a4188915#diff-3124b4902a23d9f8e06984ecf75968977214957318bf6da9ea7b8aa875d37d34L342-L345 which was aiming to add an empty ref element in newly create element for the translation.

Activating it, it creates 
![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/125b1142-78b7-4b7d-9c98-2eb8acce13cc)

So we have 2 options:
a) remove that code which was never executed
b) or restore it and see if any side effect.

I tested with it and did not noticed any obvious issue but a) is probably a better choice. What do you think @josegar74 ?